src/auth/README.md

```diff
@@ -72,69 +83,196 @@ if (blockstack.isUserSignedIn()) {
+
+This support is indicated by specifying `ES256K` in the `alg` key to indicated that the JWT signature uses ECDSA using the secp256k1 curve.  We provide both [JavaScript](https://github.com/blockstack/jsontokens-js) and [Ruby](https://github.com/blockstack/ruby-jwt-blockstack/tree/ruby-jwt-blockstack) JWT libraries with support for this curve.
+
+### Authentication request payload schema
+
+``` JavaScript
+const requestPayload = {
+    jti, // UUID
+    iat, // JWT creation time in seconds
+    exp, // JWT expiration time in seconds
+    iss, // legacy decentralized identifier generated from transit key
+    public_keys, // single entry array with public key of transit key
+    domain_name, // app origin
+    manifest_uri, // url to manifest file - must be hosted on app origin
+    redirect_uri, // url to which browser redirects user on auth approval - must be hosted on app origin
+    version, // version tuple
+    do_not_include_profile, // a boolean flag asking browser to send profile url instead of profile object
+    supports_hub_url, // a boolean flag indicating gaia hub support
+    scopes // an array of string values indicating scopes requested by the app
+  }
 ```
+ 
+  
+### Authentication response payload schema
+  
+```JavaScript
+    const responsePayload = {
+    jti, // UUID
+    iat, // JWT creation time in seconds
+    exp, // JWT expiration time in seconds
+    iss, // legacy decentralized identifier (string prefix + identity address) - this uniquely identifies the user
+    private_key, // encrypted private key payload
+    public_keys, // single entry array with public key
+    profile, // profile object or null if passed by profile_url
+    username, // blockstack id username (if any)
+    core_token, // encrypted core token payload
+    email, // email if email scope is requested & email available
+	  profile_url, // url to signed profile token
+    hubUrl, // url pointing to user's gaia hub
+    version // version tuple
+  }
+```
+  
+## `blockstack:` custom protocol handler
+
+The `blockstack:` custom protocol handler is how Blockstack apps send their authentication requests to the Blockstack Browser. When the Blockstack Browser is installed on a user's computer, it registers itself as the handler for the `blockstack:` customer protocol.
+
+When apps call [`redirectToSignIn`](http://blockstack.github.io/blockstack.js/index.html#redirecttosignin) or [`redirectToSignInWithAuthRequest`](http://blockstack.github.io/blockstack.js/index.html#redirecttosigninwithauthrequest), the code tries to check if a handler for the `blockstack:` custom protocol is installed and, if so, redirects the user to `blockstack:<authRequestToken>`. This causes the authentication request token to be passed from the app to the Blockstack Browser which in turn validates the request, displays information about the app to the user and asks the user if she wants to approve sign in to the app.
```

Comment:
When an application calls [`redirectToSignIn`](http://blockstack.github.io/blockstack.js/index.html#redirecttosignin) or [`redirectToSignInWithAuthRequest`](http://blockstack.github.io/blockstack.js/index.html#redirecttosigninwithauthrequest), `blockstack.js` checks if a `blockstack:` protocol handler is installed and, if so, redirects the user to `blockstack:<authRequestToken>`. This passes the authentication request token from the app to the Blockstack Browser, which will in turn validate the request and display an authentication dialog.